packages/hardhat-core/src/internal/cli/autocomplete.ts

```diff
@@ -1,28 +1,59 @@
+import findup from "find-up";
+import * as fs from "fs-extra";
+import * as path from "path";
+
 import { HARDHAT_PARAM_DEFINITIONS } from "../core/params/hardhat-params";
 import type hardhat from "../lib/hardhat-lib";
+import { getCacheDir } from "../util/global-dir";
```

Comment:
Should we move it to the `node_modules` dir? It will also have some edge cases, like linked projects.

Maybe we can use the global cache, but instead of saving a single file we can save one per project. What about a hash of the absolute path to the project as filename?

Reply:
The main reason I'd store them individually is that the cache can grow forever, making reading it slow.

Reply:
Done in 7fbbf4782500c4f8480b3accbc0c3014dc24f4ef